test/unit/file_transfer_agent/azure_test.js

```diff
@@ -0,0 +1,363 @@
+    result = AZ.extractContainerNameAndPath('sfc-dev1-regression//');
+    assert.strictEqual(result.containerName, 'sfc-dev1-regression');
+    assert.strictEqual(result.path, '/');
+
+    result = AZ.extractContainerNameAndPath('sfc-dev1-regression///');
+    assert.strictEqual(result.containerName, 'sfc-dev1-regression');
+    assert.strictEqual(result.path, '//');
+  });
+
+  it('get file header - success', async function () {
+    await AZ.getFileHeader(meta, dataFile);
+    assert.strictEqual(meta['resultStatus'], resultStatus.UPLOADED);
+  });
+
+  it('get file header - fail expired token', async function () {
+    mock('azure', {
+      BlobServiceClient: function() {
+        function BlobServiceClient() {
+          this.getContainerClient = function () {
+            function getContainerClient() {
+              this.getBlobClient = function () {
+                function getBlobClient() {
+                  this.getProperties = function () {
+                    function getProperties() { 
+                      this.then = function () {
+                        const err = new Error();
+                        err.code = 'ExpiredToken';
+                        throw err;
+                      };
+                    }
+                    return new getProperties;
+                  };
+                }
+                return new getBlobClient;
+              };
+            }
+            return new getContainerClient;
+          };
+        }
+        return new BlobServiceClient;
+      }
+    });
+    azure = require('azure');
+    const AZ = new SnowflakeAzureUtil(azure);
+
+    await AZ.getFileHeader(meta, dataFile);
+    assert.strictEqual(meta['resultStatus'], resultStatus.RENEW_TOKEN);
+  });
+
+  it('get file header - fail HTTP 400', async function () {
+    mock('azure', {
+      BlobServiceClient: function() {
+        function BlobServiceClient() {
+          this.getContainerClient = function () {
+            function getContainerClient() {
+              this.getBlobClient = function () {
+                function getBlobClient() {
+                  this.getProperties = function () {
+                    function getProperties() { 
+                      this.then = function () {
+                        const err = new Error();
+                        err.statusCode = '404';
+                        throw err;
+                      };
+                    }
+                    return new getProperties;
+                  };
+                }
+                return new getBlobClient;
+              };
+            }
+            return new getContainerClient;
+          };
+        }
+        return new BlobServiceClient;
+      }
+    });
+    azure = require('azure');
+    const AZ = new SnowflakeAzureUtil(azure);
+
+    await AZ.getFileHeader(meta, dataFile);
+    assert.strictEqual(meta['resultStatus'], resultStatus.NOT_FOUND_FILE);
+  });
+
+  it('get file header - fail HTTP 400', async function () {
```

Comment:
this and tests above (and some below) can be parameterized with `getProperties.then` mocked function and exepected result status?

Reply:
updated